proto/connectrpc/conformance/v2/client_compat.proto

```diff
@@ -73,6 +74,38 @@ message ClientCompatRequest {
   // For client- or bidi-streaming requests, this delay should be
   // applied before each request sent.
   uint32 request_delay_ms = 18;
+  // If present, the client should cancel the RPC instead of
+  // allowing to complete normally.
+  Cancel cancel = 19;
+  message Cancel {
+    // These fields determine the timing of cancellation.
+    // If none are present, the client should cancel immediately
+    // after all request messages are sent the stream half-closed
```

Comment:
Can you elaborate on what you mean by `half-close`? Does that mean closing the send side for client/bidi streams and closing the receive side for server streams?

Also, just to confirm I'm following:

- `before_half_close` example would be a client or bidi stream sends all its requests and then cancels (in Go parlance, we'd cancel the context) and then closes the send side? 
- `after_half_close_ms` example would be a server stream sends the request, closes the send side of the stream, waits for `x` ms and then cancels?

Reply:
"Half-close" means the client closes the request-ending side. The other "half", the server sending responses, remains open until the stream is complete.

In grpc-go, this operation is called "close send". In connect-go, it's called "close request", In the HTTP/2 specs, this state is called "half-closed".

> `before_half_close` example would be a client or bidi stream sends all its requests and then cancels (in Go parlance, we'd cancel the context) and then closes the send side?

Mostly right. After cancelling, there should be no need to close the send side: cancelling should implicitly close everything.

> `after_half_close_ms` example would be a server stream sends the request, closes the send side of the stream, waits for `x` ms and then cancels?

Exactly.

Reply:
I guess I can rename these `before_close_send` and `after_close_send_ms`?

Reply:
Either way is fine. Let's stick with `half_close` but maybe just add a bit more detail on what that actually means in the comments?

Reply:
Doh. I just renamed it and updated the docs 🤦.

Maybe take another look to see if you like the updates?

Reply:
Looks great. Thanks!